src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -1483,6 +1484,91 @@ def subnet_role_assignment_exists(cli_ctx, scope):
     return False
 
 
+def aks_check_acr(cmd, client, resource_group_name, name, acr):
+    if not which("kubectl"):
+        raise CLIError("Can not find kubectl executable in PATH")
```

Comment:
CLIError is not recommended to use.
https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md

Reply:
Update the error to a ValidationError